### PR TITLE
Accept 10-10CG Submissions without a Primary Caregiver

### PR DESCRIPTION
**Pull request by kevinmirc**
Fixes: https://github.com/department-of-veterans-affairs/va.gov-team/issues/16454
Related Schema PR: https://github.com/department-of-veterans-affairs/vets-json-schema/pull/537

**Description**
Add support for the [latest 10-10CG schema changes]() which allow submissions without a primary caregiver.

**Summary**
- [x] Remove all direct references to Primary Caregiver.
- [x] Remove all assumptions that Primary Caregiver will be present.
- [x] Retire `CARMA::Models::Base` in favor of a more explicit, hard-coded, `#to_request_payload` on each inheriting model.
  - Will implement in separate PR: https://github.com/department-of-veterans-affairs/va.gov-team/issues/16129
- [x] Mock calls to `Metadata` in `spec/lib/carma/models/submission_spec.rb` to avoid double testing `Metadata` and causing unit tests for `Submission` to execute `Metadata` source code.
  - Will implement in separate PR: https://github.com/department-of-veterans-affairs/va.gov-team/issues/16130

**Breaking Changes** to the CARMA data contract ⚠️ 
- `data.primaryCaregiver` is now **optional**.
  - [x] CARMA has been notified.
  - [x] Dataflow has been tested (Local <> SIT).

- `data.secondaryCaregiverOne` is now **optional**.
  - [x] CARMA has been notified.
  - [x] Dataflow has been tested (Local <> SIT).

- One of `data.primaryCaregiver` or `data.secondaryCaregiverOne` is guaranteed to be present.
  - [x] CARMA has been notified.
  - [x] Dataflow has been tested (Local <> SIT).

- One of `metadata.primaryCaregiver` or `metadata.secondaryCaregiverOne` is guaranteed to be present when found in `data.{subject}`.
  - [x] CARMA has been notified.
  - [x] Dataflow has been tested (Local <> SIT).

_Testing results found here_: https://dsva.slack.com/archives/CUVNE9Q2E/p1605204939006400

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -109,7 +109,7 @@ gem 'statsd-instrument', '~> 2.6.0' # versions beyond 2.6 deprecate config and c
 gem 'swagger-blocks'
 gem 'typhoeus'
 gem 'utf8-cleaner'
-gem 'vets_json_schema', git: 'https://github.com/department-of-veterans-affairs/vets-json-schema', branch: 'master'
+gem 'vets_json_schema', git: 'https://github.com/department-of-veterans-affairs/vets-json-schema', branch: '10-10cg-new-version'
```
Comment (kfrz) [inline Gemfile:112]: What's the reason this change isn't merged into `master` on `vets-json-schema` ?

File: lib/carma/README.md [hunk 2]
```diff
@@ -2,9 +2,9 @@
 CARMA (Caregiver Record Management Application)
 
 ## Description
-CARMA (Caregiver Record Management Application) is a Salesforce application that the VA's Caregivers Program uses to intake, track, and process 10-10CG submissions.
+CARMA is a Salesforce application that the VA's Caregivers Program uses to track and process 10-10CG submissions.
 
-This CARMA service/module is used to submit valid, online, 10-10CG submissions (CaregiversAssistanceClaim) to CARMA. It includes models that map to the Salesforce API interface and an http client.
+This library is used to submit valid, online, 10-10CG submissions (CaregiversAssistanceClaim) to CARMA. It includes an http client and models that map to the CARMA domain.
```
Comment (kfrz) [inline lib/carma/README.md:7]:
```suggestion
This library is used to submit valid, online, 10-10CG submissions (CaregiversAssistanceClaim) to CARMA. It includes an HTTP client and models that map to the CARMA domain.
```
minor nitpick

File: lib/carma/models/metadata.rb [hunk 3]
```diff
@@ -37,19 +37,23 @@ def veteran=(veteran_data_hash)
...
+        @primary_caregiver  = if pc_metadata_hash.nil? # rubocop:disable Layout/SpaceAroundOperators, Layout/ExtraSpacing
+                                nil
+                              else
+                                Caregiver.new(pc_metadata_hash)
+                              end
       end
 
       def secondary_caregiver_one=(sc_one_metadata_hash)
-        @secondary_caregiver_one =  if sc_one_metadata_hash.nil?
+        @secondary_caregiver_one  = if sc_one_metadata_hash.nil? # rubocop:disable Layout/SpaceAroundOperators
```
Comment (kfrz) [inline lib/carma/models/metadata.rb:48]: Why are the extra spaces/rubocop disable necessary?

**Review by kfrz — COMMENTED** [review]
Overall these changes look fine, functionally, I have some small nitpick comments.

File: Gemfile (hunk 1 above)
Comment (kevinmirc) [reply Gemfile:112]: I reference the `10-10cg-new-version` branch to ensure that the new schema changes will work with this version of vets-api code. I'll be changing this vets_json_schema reference back to master before merging (after the [schema changes are merged to master](https://github.com/department-of-veterans-affairs/vets-json-schema/pull/537)).

File: lib/carma/models/metadata.rb (hunk 3 above)
Comment (kevinmirc) [reply lib/carma/models/metadata.rb]: I was struggling to satisfy the rubocop rules for aligning `if` and `else` while keeping indentation at increments of 2 spaces. After another attempt, I was able to pass rubocop by using an odd number of spaces for indentation and keeping the conditional statement aligned. Does that work?

File: Gemfile (hunk 1 above)
Comment (kfrz) [reply Gemfile:112]: Ok, awesome. That PR is approved, so once merged this line will change to reference the `master` branch. Thanks for clarifying!

File: lib/carma/README.md [hunk 4]
```diff
@@ -45,7 +45,7 @@ submission    = CARMA::Models::Submission.from_claim(
                   }
                 )
 
-submission.submit!(carma_client)
+submission.submit!(carma_client)
```
Comment (kfrz) [inline lib/carma/README.md:48]:
```suggestion
submission.submit!(carma_client)
```

Remove trailing whitespace

File: Gemfile (hunk 1 above)
Comment (kfrz) [inline Gemfile:112]:
```suggestion
gem 'vets_json_schema', git: 'https://github.com/department-of-veterans-affairs/vets-json-schema', branch: 'master'
```

Reference `master` branch

Comment (kfrz) [reply Gemfile:112]: Adding a suggested commit so you can make the change & not require re-review

**Review by kfrz — APPROVED** [review]
Approving with a suggested commit for reverting to `master` on the schema!

File: Gemfile.lock [hunk 5]
```diff
@@ -65,10 +65,10 @@ GIT
 
 GIT
   remote: https://github.com/department-of-veterans-affairs/vets-json-schema
-  revision: 4f2d386700245582301ae2b04ddd629e422697a0
-  branch: master
+  revision: 17c2992bce1572a67490d76193e0568177810399
+  branch: 10-10cg-new-version
```
Comment (annaswims) [inline Gemfile.lock:69]: I don't think it's ok to run vets-json-schema on non-master.

Comment (kevinmirc) [reply Gemfile.lock:69]: This was a mistake, fix here: https://github.com/department-of-veterans-affairs/vets-api/pull/5371

**Pull request merged**